Assets/Scripts/Player/PlayerControls.cs

```diff
@@ -425,22 +411,33 @@ void SetupStateMethodMap()
 		_actor.physics.RegisterState( PhysicsStateType.Grounded,  new Grounded( _actor ) );
 		_actor.physics.RegisterState( PhysicsStateType.Climbing,  new Climbing( _actor ) );
 		_actor.physics.RegisterState( PhysicsStateType.Gliding,   new Gliding( _actor ) );
-		_actor.physics.RegisterState( PhysicsStateType.Sprinting, new Sprinting( _actor ) );
 	}
 	#endregion
 
 	public bool InteractCheck( RaycastHit hitInfo )
 	{
-		Interactable interactable = hitInfo.collider.gameObject.GetComponent<Interactable>();
+		Interactable interactable = hitInfo.collider.GetComponent<Interactable>();
 		if ( interactable )
 		{
 			_textBox.SetTextForDuration( interactable.interactText, interactable.duration );
 			return true;
 		}
-		else
+
+		return false;
+	}
+
+	public bool AdultBuddyCheck( RaycastHit hitInfo )
+	{
+		Debug.Log( "Checking if we can make buddy an adult." );
+		BuddyStats buddyStats = hitInfo.collider.GetComponentInParent<BuddyStats>();
+		if ( buddyStats && buddyStats.isAdult )
 		{
-			return false;
+			// TODO: How do we start the process of actually making the buddy an adult?

```

Comment:
Nope. That couldn't be completed until we had the buddy-on-back functionality, which wasn't in at the time.